src/Agent.Worker/DiagnosticLogManager.cs

```diff
@@ -346,5 +353,37 @@ private string GetEnvironmentContentNonWindows(int agentId, string agentName, IL
 
             return builder.ToString();
         }
+
+        private async Task<string> DumpEventLogs(DateTime jobStartTimeUtc, string diagFolder){
+            var builder = new StringBuilder();
+
+            string powerShellExe = HostContext.GetService<IPowerShellExeUtil>().GetPath();
```

Comment:
Let's catch exceptions here and print error message - so any issues with logs dumping won't break the agent itself.

Reply:
Done